tools/stonith_admin.c

```diff
@@ -553,10 +553,20 @@ main(int argc, char **argv)
         crm_bump_log_level(argc, argv);
     }
 
+    rc = pcmk__output_new(&out, output_ty, output_dest, argv);
+    if (rc != 0) {
+        fprintf(stderr, "Error creating output format %s: %s\n", output_ty, pcmk_strerror(rc));
+        exit_code = CRM_EX_ERROR;
+        goto done;
+    }
+
+    stonith__register_messages(out);
+
     if (options.version) {
-        fprintf(stdout, "Pacemaker %s\n", PACEMAKER_VERSION);
-        fprintf(stdout, "Written by Andrew Beekhof\n");
-        crm_exit(CRM_EX_OK);
+        out->info(out, "Pacemaker %s", PACEMAKER_VERSION);
+        out->info(out, "Written by Andrew Beekhof");
```

Comment:
I think a way to get the pacemaker version in XML would be handy, so perhaps a formatted message would be appropriate. However I'm not sure whether that should be a custom message or one of the base methods; we could leave that for later.

Reply:
I could go either way with it, but if you want a tiebreaker, there is currently nowhere to put custom messages that all the tools can access.  I could add something.

Reply:
We can leave it for later. I tend to lean toward putting anything that would go in libcrmcommon as full-fledged methods rather than custom messages.